### DISCUSSION TITLE
Is these any Request ID in flask request object

### DISCUSSION THREAD

**Discussion by kashcode** [pallets/flask#5435] [Q&A]
I wondering does Flask have some property, what can be accessed and logged, by that later on error all request can be traced down from start till end?

**Comment by RonnyPfannschmidt** [comment]
Flask has no own request I'd mechanism

Are you looking for open tracing integration?

**Reply by kashcode** [reply]
I have my own logging mechanism and it will be cool if I can aggregate all actions under some request identifier, and I start thinking how it can be done and came across Request ID mechanism.

btw ... what you mean by "open tracing" - I don't need anything complex

**Comment by jtrip** [comment]
As Ronny mentioned there is nothing in Flask that does that already, but depending on your needs it could be very easy to implement. You could just generate a unique id in an `@app.before_request` decorated function and store it on `g` and then log it whenever you like.
